engine/engine.py

```diff
@@ -115,14 +115,29 @@ def _unsafe_check_neighbors(self, x, y):
 
     def _check_safe_first_move(self, x, y):
         nb = self._get_neighboring_bombs(x, y)
-        while nb > 0:
+
+        if nb > 0:
             self._clear_neighboring_bombs(x, y)
-            new_coords = np.floor(np.multiply(np.random.rand(nb, 2), np.array([[self._width, self._height]]))).astype(int)
-            for nc in new_coords:
-                while self._board[nc[0], nc[1]] == Spaces.BOMB:
-                    nc = np.floor(np.multiply(np.random.rand(1, 2), np.array([[self._width, self._height]]))).astype(int)[0]
-                self._board[nc[0], nc[1]] = Spaces.BOMB
-            nb = self._get_neighboring_bombs(x, y)
+
+            empty_tiles = []
+
+            for x in range(self._width):
```

Comment:
I hate to say it but could you add a unit test to make sure the number of bombs is correct after this runs?

Reply:
It seems like with this logic, the original space could be picked as a bomb again since you cleared the neighbors beforehand

Reply:
Oh right duh.

Reply:
That should be fixed now